docs/website/01_theory/controlRoutines.md

```diff
@@ -28,3 +28,116 @@ enum class ControlMessages {
   SOLUTION_CHANGED,
   END };
 ```
+
+## Path following techniques
+A general routine based on the standard Arc-length method is included which uses a scalar subsidiary function to impose 
+a constraint on the non-linear system of equations. The previously mentioned LoadControl method can also be recreated 
+using this technique. For more details on the standard arc-length method, refer the works 
+of Wempner[@WEMPNER19711581], Crisfield[@CRISFIELD198155], Ramm[@wunderlich_strategies_1981] and 
+Riks[@riks_application_1972] among others. A path following object is constructed as follows:  
+```cpp
+auto alc = Ikarus::PathFollowing(nr, load_steps, stepSize, pft);
+```
+where `#!cpp nr` is a Newton-Raphson solver which considers a scalar subsidiary function and is defined by
+```cpp
+auto nr = Ikarus::makeNewtonRaphsonWithSubsidiaryFunction(nonLinOp, std::move(linSolver));
+```
+and `#!cpp pft` is the desired path following technique. Three different path following techniques are included, namely
+
+* Standard arc-length method
+* Load control method (as a subsidiary function under this generalized implementation)
+* Displacement control method (uses a vector of indices which are all controlled by a same `#!cpp stepSize`).
+
+These can be invoked by defining 
+```cpp
+auto pft = Ikarus::StandardArcLength{psi};
+auto pft = Ikarus::LoadControlWithSubsidiaryFunction{};
+auto pft = Ikarus::DisplacementControl{controlledIndices};
+```
+The default path following type is the `#!cpp Ikarus::StandardArcLength{psi=1.0}`.
+
+It is to note that in the current implementation, it is assumed that the external forces are given by 
+$F_{ext} = F_{ext}^0\lambda$ such that 
+$$
+-\frac{\partial \mathbf{R}}{\partial \lambda} = F_{ext}^0
+$$
+An implementation for a general non-linear $F_{ext} = F_{ext}^0\left(\mathbf{D},\lambda\right)\lambda$ is an open task.
+
+In order to create an own implementation for the scalar subsidiary function, the user has to create a `#!cpp struct` 
+with the following three functions:  
+```cpp
+void evaluateSubsidiaryFunction(SubsidiaryArgs& args);
+void initialPrediction(NonLinearOperator& nonLinearOperator, SubsidiaryArgs& args);
+void intermediatePrediction(NonLinearOperator& nonLinearOperator, SubsidiaryArgs& args);
+```
+The function `#!cpp evaluateSubsidiaryFunction(SubsidiaryArgs& args)` is used to evaluate the subsidiary function and 
+its derivatives with respect to the displacement $\mathbf{D}$ and the load factor $\lambda$ for every Newton-Raphson iteration. The other two functions 
+are used to specify a prediction for $\mathbf{D}$ and $\lambda$ initially for the very first `#!cpp load_steps` and for 
```

Comment:
Instead of writing `#!cpp load_steps`, i think we can just write "for the very first step"